bciers/apps/registration/tests/components/form/widgets/MultiSelectWidget.test.tsx

```diff
@@ -0,0 +1,269 @@
+  });
+
+  it("should allow selecting a combo box value", async () => {
+    render(
+      <FormBase
+        schema={multiSelectFieldSchema}
+        uiSchema={multiSelectFieldUiSchema}
+      />,
+    );
+
+    const openMultiSelectButton = screen.getByRole("button", { name: "Open" });
+    await userEvent.click(openMultiSelectButton);
+
+    const option3 = screen.getByText("Option 3");
+
+    await userEvent.click(option3);
+
+    expect(screen.getByRole("button", { name: "Option 3" })).toBeVisible();
+  });
+
+  it("should allow selecting multiple combo box values", async () => {
+    render(
+      <FormBase
+        schema={multiSelectFieldSchema}
+        uiSchema={multiSelectFieldUiSchema}
+      />,
+    );
+
+    const openMultiSelectButton = screen.getByRole("button", { name: "Open" });
+    await userEvent.click(openMultiSelectButton);
+
+    const option1 = screen.getByText("Option 1");
+    const option3 = screen.getByText("Option 3");
+
+    await userEvent.click(option1);
+
+    await userEvent.click(openMultiSelectButton);
+
+    await userEvent.click(option3);
+
+    expect(screen.getByText("Option 1")).toBeVisible();
+    expect(screen.getByText("Option 3")).toBeVisible();
+  });
+
+  it("should allow typing a combo box value", async () => {
+    render(
+      <FormBase
+        schema={multiSelectFieldSchema}
+        uiSchema={multiSelectFieldUiSchema}
+      />,
+    );
+
+    const multiSelectInput = screen.getByRole("combobox") as HTMLInputElement;
+
+    await userEvent.type(multiSelectInput, "Option 3{enter}");
+
+    expect(screen.getByRole("button", { name: "Option 3" })).toBeVisible();
+  });
+
+  it("should allow clearing the combo box", async () => {
+    render(
+      <FormBase
+        schema={multiSelectFieldSchema}
+        formData={{ multiSelectTestField: ["option_2", "option_3"] }}
+        uiSchema={multiSelectFieldUiSchema}
+      />,
+    );
+
+    const selectedOption2 = screen.getByRole("button", { name: "Option 2" });
+    const selectedOption3 = screen.getByRole("button", { name: "Option 3" });
+    const optionCancelIcon = screen.getAllByTestId("CancelIcon");
+
+    await userEvent.click(optionCancelIcon[0]);
+    await userEvent.click(optionCancelIcon[1]);
+
+    waitFor(() => {
+      expect(selectedOption2).not.toBeInTheDocument();
+      expect(selectedOption3).not.toBeInTheDocument();
+    });
+  });
+  it("should render the  placeholder text", async () => {
+    render(
+      <FormBase
+        schema={multiSelectFieldSchema}
+        uiSchema={{
+          multiSelectTestField: {
+            "ui:widget": "MultiSelectWidget",
+            "ui:placeholder": "Select regulated products",
+          },
+        }}
+      />,
+    );
+    expect(
+      screen.getByPlaceholderText("Select regulated products..."),
+    ).toBeVisible();
+  });
+  it("should use the enum values as the names if no enumNames are provided, use the enum values as the names", async () => {
```

Comment:
this test name seems a little repetitive, a little bit

Reply:
good point point